src/Analyzer/Utils.h

```diff
@@ -102,4 +102,12 @@ NameSet collectIdentifiersFullNames(const QueryTreeNodePtr & node);
 /// Wrap node into `_CAST` function
 QueryTreeNodePtr createCastFunction(QueryTreeNodePtr node, DataTypePtr result_type, ContextPtr context);
 
+/// Resolves function node as ordinary function with given name.
+/// Arguments and parameters are taken from the node.
+void resolveOrdinaryFunctionNodeByName(FunctionNode & function_node, const String & function_name, const ContextPtr & context);
+
+/// Resolves function node as aggregate function with given name.
+/// Arguments and parameters are taken from the node.
+void resolveAggregateFunctionNodeByName(FunctionNode & function_node, const String & function_name);
```

Comment:
Why does it require to pass name? If you change the name of the function, `FunctionNode::getFunctionName` will still return the old one.

Reply:
But after call of `FunctionNode::resolveAsAggregateFunction` it will return [new name](https://github.com/ClickHouse/ClickHouse/blob/6e1465af10cd217dab31619f9e63398a2685223a/src/Analyzer/FunctionNode.cpp#L85). Also I removed usages of `function_node.getFunctionName()` from `resolveAggregateFunction` in `Utils.cpp`.